workers/unity/Packages/io.improbable.gdk.core/.codegen/Source/Generators/Core/UnityComponentDataGenerator.cs

```diff
@@ -100,33 +100,14 @@ component MyComponent
                         }
```

Comment:
The check and exception above this, should this also be in `ValidateFieldIndex`?